### DISCUSSION TITLE
in-app pure frontend notification

### DISCUSSION THREAD

**Discussion by iambighead** [wailsapp/wails#2015] [Q&A]
I am working on latest wails 2.1. 
Not sure if this has been discussed before, but I am looking for a way to have simple in-app notification, like for info, or warning, or error, something better than plain alert ()

I see there is a road map for Notification support  https://github.com/wailsapp/wails/issues/1788
but that seems to from golang backend to OS. I am looking from pure front-end perspective. 

I have tried things like noty, reach-notification, react-toastify, but they don't seem to work for me.
I have experience with noty on pure browser/front-end development before so I am pretty sure it should works, but not sure if there is something i need to handle to get it working with wails app.

I am not looking to add additional component, i am happy to use 3rd part lib as long as it works.

Any one has success adding notification to their wails app?

**Comment by alanchenchen** [comment]
According to MDN article，Https procotal is necessary for Web Notification。
Here is it：
![image](https://user-images.githubusercontent.com/23295628/199147883-b8293e91-5dcb-4766-9927-0ca5056a4694.png)

In Wails app，it's static file but no server.Maybe pure frontend  notification is not a good way

**Reply by iambighead** [reply]
i guess you are saying something like web push notification
I am actually referring to just an alert popup within wails app only, there is no network messaging needed.

**Reply by alanchenchen** [reply]
Maybe native GUI renderer is needed. I've seen the roadmap, wails notification API should wait for next version to resolve cross-platform compatibility.

**Reply by leaanthony** [reply]
Notifications is on the roadmap @iambighead

**Reply by iambighead** [reply]
@leaanthony thanks. May i check the notification being mentioned is a desktop, aka OS level native notification, or in-app notification (render by HTML/CSS) ? I think Desktop / OS level notification are good and necessary, but currently I am look at a in-app notification mechanism.

**Reply by leaanthony** [reply]
Oh, if you want a toast-like notification there's plenty of libraries to do that.

**Reply by iambighead** [reply]
yeah thats what I though as well but i tried a few seems not working me, so i am wondering if I am doing something wrong LOL
disclaimer I am pretty new to wails app development, but I had some experience from Electron.
